DllImportGenerator/TestAssets/Benchmarking/CodeGenMonitor.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Benchmarking.Visualizer;
+
+namespace Benchmarking
+{
+    /// <summary>
+    /// Data about the method and its generated code.
+    /// </summary>
+    public record MethodCodeGen
+    {
+        public uint MetadataToken { get; init; }
+        public bool IsILStub { get; init; }
+        public string FullyQualifiedClassName { get; init; }
+        public string MethodName { get; init; }
+        public string GeneratedCode { get; init; }
+        public uint GeneratedCodeSize { get; init; }
+        public string ILCode { get; init; }
+    }
+
+    /// <summary>
+    /// Class used to monitor code generation in the CLR
+    /// </summary>
+    public class CodeGenMonitor : IDisposable
+    {
+        public const int MetadataTokenNil = 0;
+
+        private readonly CodeGenEventListener listener;
+        private readonly int retryDelayInMs;
+        private readonly HtmlCompareTwoPane htmlCompare;
+
+        private bool isDisposed = false;
+
+        private Dictionary<string, List<MethodCodeGen>> methods = new Dictionary<string, List<MethodCodeGen>>(StringComparer.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// Initialize a <see cref="CodeGenMonitor"/> instance.
+        /// </summary>
+        /// <param name="retryDelayInMs">Time delay (ms) to use when requesting a function name that isn't immediately known</param>
+        public CodeGenMonitor(int retryDelayInMs = 20)
+        {
+            this.listener = new CodeGenEventListener();
+            this.listener.NewMethodCodeGen += NewMethodCodeGen;
+            this.retryDelayInMs = retryDelayInMs;
+
+            this.htmlCompare = new HtmlCompareTwoPane()
+            {
+                Title = "Compare generated code",
+                Pane1Title = "Code Gen 1",
+                Pane2Title = "Code Gen 2",
+                SelectionTitle = "All methods",
+            };
+        }
+
+        /// <summary>
+        /// Get the generated code for the supplied method.
+        /// </summary>
+        /// <param name="fullyQualifiedMethodName">Fully qualified method name</param>
+        /// <param name="metadataToken">[Optional] method's metadata token - used to clarify overloads.</param>
+        /// <param name="retryCount">Number of times to retry if method isn't immediately found</param>
+        /// <returns>A <see cref="MethodCodeGen"/> instance</returns>
+        public MethodCodeGen GetLastCodeGenFor(string fullyQualifiedMethodName, int metadataToken = MetadataTokenNil, int retryCount = 5)
```

Comment:
Do we really need this? If the actual implementation is async, I'd much rather avoid any sort of synchronous wrapper and just have the async API.